ql/src/java/org/apache/hadoop/hive/ql/metadata/VirtualColumn.java

```diff
@@ -50,10 +50,14 @@
 
   RAWDATASIZE("RAW__DATA__SIZE", TypeInfoFactory.longTypeInfo),
   /**
-   * {@link org.apache.hadoop.hive.ql.io.RecordIdentifier} 
+   * {@link org.apache.hadoop.hive.ql.io.RecordIdentifier}
    */
   ROWID("ROW__ID", RecordIdentifier.StructInfo.typeInfo, true, RecordIdentifier.StructInfo.oi),
   ROWISDELETED("ROW__IS__DELETED", TypeInfoFactory.booleanTypeInfo),
+  PARTITION_SPEC_ID("PARTITION__SPEC__ID", TypeInfoFactory.intTypeInfo),
+  PARTITION_HASH("PARTITION__HASH", TypeInfoFactory.longTypeInfo),
+  FILE_PATH("FILE__PATH", TypeInfoFactory.stringTypeInfo),
```

Comment:
Isn't this the same as the `INPUT__FILE__NAME` in the delete case?

Reply:
Unfortunately no, the `INPUT__FILE__NAME` seems to be used differently and `ctx.getCurrentInputPath()` (where its value comes from) only contains the root location, not the full file path:
```
file:/Users/martonbod/Repos/Hive/Fork/hive/iceberg/iceberg-handler/target/tmp/hive5438133120236603811/external/customers
```